File: src/main/kotlin/com/example/kotlinspring/controller/TrainerController.java
package com.example.kotlinspring.controller;

import com.example.kotlinspring.entity.Trainer;
import com.example.kotlinspring.repository.TrainerRepository;
import jakarta.persistence.NoResultException;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController("trainer")
public class TrainerController {

    @Autowired
    private TrainerRepository trainerRepository;

    @GetMapping("/")
    public List<Trainer> getAll(){
        return trainerRepository.findAll();
    }

    @GetMapping("/{id}")
    public Trainer getById(Long id){
        return trainerRepository.findById(id).orElseThrow(() -> new NoResultException("Trainer with id: " + " not found."));
    }

    @PostMapping("/")
    public Trainer add(Trainer trainer){
        return trainerRepository.save(trainer);
    }
}

File: src/main/kotlin/com/example/kotlinspring/controller/TrainerController.kt
package com.example.kotlinspring.controller

import com.example.kotlinspring.entity.Trainer
import com.example.kotlinspring.repository.TrainerRepository
import jakarta.persistence.NoResultException
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/trainer")
class TrainerController {

    @Autowired
    private lateinit var trainerRepository: TrainerRepository

    @GetMapping("/")
    fun getAll(): List<Trainer> {
        return trainerRepository.findAll()
    }

    @GetMapping("/{id}")
    fun getById(@PathVariable id: Long): Trainer {
        return trainerRepository.findById(id).orElseThrow { NoResultException("Trainer with id: $id not found.") }
    }

    @PostMapping("/")
    fun add(@RequestBody trainer: Trainer): Trainer {
        return trainerRepository.save(trainer)
    }
}
